### DISCUSSION TITLE
Explore improving UX around interaction of console messages and breakpoints

### DISCUSSION THREAD

**Discussion by bhackett1024** [RecordReplay/devtools#4623] [General]
In this loom which @Andarist created https://www.loom.com/share/da5010eec37041e3a01f4c6d9eb9eba2, at around 2:40 there is an issue where after jumping to a console message, it's possible to rewind to a breakpoint on the same line as the message, even though in the UI it seems like we're paused at the same place.  This happens because jumping to a console message brings the browser to the console.log call itself, which is a different place from the start of the line which logged the message (which is where we are after setting a breakpoint).  This is confusing, and it would be good to think about ways the UX could be improved here.  One option would be to find a way to distinguish the two cases with an icon or something when we are inside the console message, another option would be to find a simple way of flattening these two points together in the UI when stepping / using breakpoints, and maybe there are others.

**Comment by jasonLaster** [comment]
Removing from planning since this is more of a general UI polish which we should get to when we have cycles than a project that relates to launch

**Comment by jaril** [comment]
This seems somewhat related to #3818 (i.e. multiple breakable points on the same line) so just linking that there so it's noted. It's still different enough that it should be its own ticket.

**Comment by jasonLaster** [comment]
Would love to get thoughts / suggestions from the community

**Comment by jonbell-lot23** [comment]
March 15, 2022:

Agree that this is confusing. It's a real humdinger.
